feat: Current order is now editable

alecorvec opened the pull request:
## Describe your changes

Added new buttons for each foods, used to either increase or decrease the quantity of a food, and a button used to modify the food itself

## How to test the feature

Add a food to the order. Click on the food name on the right, buttons should appear. Clicking on "+" will increase food quantity. Clicking on "-" will decrease quantity, and if 0 is reached, will remove food from the order. Clicking on "Modifier" will take yo to that food page, with the choices already made. Change the choices, click on "Modifier" (at the bottom of the screen), the food will update.

## Issue ticket number and link

#102 

## Checklist before requesting a review
- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have performed a self-review of my code
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Documentation has been updated as required
If a single item in this checklist is not checked, the pull request cannot be accepted

JulesGresset reviewed (changes requested):
Feature works as intended for newly created orders, but :
- [x] When an order is recalled (from the "COMMANDES" tab for instance), the name of the food is not shown
- [x] When quantity is modified, the total price of the order is not updated

alecorvec commented:
>     * [ ]  When an order is recalled (from the "COMMANDES" tab for instance), the name of the food is not shown

The issue is now fixed : It was because the orders object used in the POS rendering has different variables than the object returned from the API call in the command tab (mainly missing the "number" variable, used to render the quantity of a food. If 0 or Undefined, makes the name disappear, causing the bug). Changed the "part" variable from the API object to "number". If part isn't the correct variable for this, don't merge this PR (couldn't find documentation on this specific variable).

JulesGresset reviewed (changes requested):
Now the bug that makes the name of a food disappear is fixed.
One thing still, you can successfully edit the quantity or details of a food in a newly created order but for a recalled order (from the 'COMMANDES' tab), you can't edit a food.
Also, the part that sends the total amount of the order that's been merged in PR #137 seems to be broken now (works on dev)
